packages/core/core/src/requests/WriteBundlesRequest.js

```diff
@@ -0,0 +1,181 @@
+// @flow strict-local
+
+import type {Async} from '@parcel/types';
+import type {SharedReference} from '@parcel/workers';
+import type {StaticRunOpts} from '../RequestTracker';
+import type {ContentKey, PackagedBundleInfo} from '../types';
+import type BundleGraph from '../BundleGraph';
+import type {BundleInfo} from '../PackagerRunner';
+
+import {HASH_REF_PREFIX} from '../constants';
+import {serialize} from '../serializer';
+import nullthrows from 'nullthrows';
+import path from 'path';
+import {hashString} from '@parcel/hash';
+import {createPackageRequest} from './PackageRequest';
+import createWriteBundleRequest from './WriteBundleRequest';
+
+type WriteBundlesRequestInput = {|
+  bundleGraph: BundleGraph,
+  optionsRef: SharedReference,
+|};
+
+type RunInput = {|
+  input: WriteBundlesRequestInput,
+  ...StaticRunOpts,
+|};
+
+export type WriteBundlesRequest = {|
+  id: ContentKey,
+  +type: 'write_bundles_request',
+  run: RunInput => Async<Map<string, PackagedBundleInfo>>,
+  input: WriteBundlesRequestInput,
+|};
+
+/**
+ * Packages, optimizes, and writes all bundles to the dist directory.
+ */
+export default function createWriteBundlesRequest(
```

Comment:
Is there a better name for this request? It's very similar to the singular `WriteBundleRequest`. This one does the entire process of packaging, optimizing, and writing every bundle. Open to name suggestions here.